instrumentation/opentelemetry-instrumentation-kafka-python/src/opentelemetry/instrumentation/kafka/utils.py

```diff
@@ -55,27 +55,36 @@ def extract_send_headers(args, kwargs):
     @staticmethod
     def extract_send_partition(instance, args, kwargs):
         """extract partition `send` method arguments, using the `_partition` method in KafkaProducer class"""
-        topic = KafkaPropertiesExtractor.extract_send_topic(args)
-        key = KafkaPropertiesExtractor.extract_send_key(args, kwargs)
-        value = KafkaPropertiesExtractor.extract_send_value(args, kwargs)
-        partition = KafkaPropertiesExtractor._extract_argument(
-            "partition", 4, None, args, kwargs
-        )
-        key_bytes = instance._serialize(
-            instance.config["key_serializer"], topic, key
-        )
-        value_bytes = instance._serialize(
-            instance.config["value_serializer"], topic, value
-        )
-        valid_types = (bytes, bytearray, memoryview, type(None))
-        if (
-            type(key_bytes) not in valid_types
-            or type(value_bytes) not in valid_types
-        ):
+        try:
+            topic = KafkaPropertiesExtractor.extract_send_topic(args)
+            key = KafkaPropertiesExtractor.extract_send_key(args, kwargs)
+            value = KafkaPropertiesExtractor.extract_send_value(args, kwargs)
+            partition = KafkaPropertiesExtractor._extract_argument(
+                "partition", 4, None, args, kwargs
+            )
+            key_bytes = instance._serialize(
+                instance.config["key_serializer"], topic, key
+            )
+            value_bytes = instance._serialize(
+                instance.config["value_serializer"], topic, value
+            )
+            valid_types = (bytes, bytearray, memoryview, type(None))
+            if (
+                type(key_bytes) not in valid_types
+                or type(value_bytes) not in valid_types
+            ):
+                return None
+
+            all_partitions = instance._metadata.partitions_for_topic(topic)
+            if all_partitions is None or len(all_partitions) == 0:
+                return None
+
+            return instance._partition(
+                topic, partition, key, value, key_bytes, value_bytes
+            )
+        except Exception as exception:  # pylint: disable=W0703
```

Comment:
How does this affect users? In case this case does trigger, will we end up not recording a span, omitting some info from the span or will this affect the instrumented service or kafka client in some way?

Reply:
It will affect the instrumented service - hence the need to protect it with try/except

Reply:
I understand that. I'm asking how does this fix affect users? Will it result in missing spans, missing attributes on spans or something else entirely?

Reply:
Missing attribute - the partition